pkg/etcd/etcd.go

```diff
@@ -217,6 +248,57 @@ func (e *Client) Watch(ctx context.Context, prefix string, revision int64) clien
 	return e.client.Watch(ctx, prefix, clientv3.WithPrefix())
 }
 
+// DoTxn does some operation in one transaction.
+// Note: should only have one opereration for one key, otherwise will get duplicate key error.
+func (e *Client) DoTxn(ctx context.Context, operations []*Operation) (int64, error) {
+	cmps := make([]clientv3.Cmp, 0, len(operations))
+	ops := make([]clientv3.Op, 0, len(operations))
+
+	for _, operation := range operations {
+		operation.Key = keyWithPrefix(e.rootPath, operation.Key)
+
+		if operation.TTL > 0 {
```

Comment:
should we need to check `operation.Tp != DeleteOp` here because `WithLease` only takes effect for `PUT` operation?

Reply:
yes, if set TTL for delete,  etcd client will panic `Panic: unexpected lease in delete`, I add check in af2a844